### DISCUSSION TITLE
SD card sorting

### DISCUSSION THREAD

**Discussion by mrga37** [Jyers/Marlin#1725] [Q&A]
I was wondering if the SD card on printer can be sorted so that I get, for example, name sorting, date sorting etc.

**Comment by K2Kevin** [comment]
> I was wondering if the SD card on printer can be sorted so that I get, for example, name sorting, date sorting etc.

That would be nice, see the files I just copied to card on top (Modified Date) would be a nice option.

**Reply by jumper-11** [reply]
Hello,
has anyone solved this problem? Is there any improved soft?
I have version 2.0.2h now and this is a problem.
This is very annoying.

**Reply by jumper-11** [reply]
why no one will correct it? It is very tedious.

**Reply by jumper-11** [reply]
nothing is impossible :)
I do not believe that the author does not care about improving the quality of the firmware

**Reply by jumper-11** [reply]
Then why the original soft can do it?
On the original software - I put in the project card and the latest files are from the top. If others ask about it, it means that it was so and suddenly it is not. So there is no technical obstacle here.

**Reply by brunohenriquy** [reply]
> @jumper-11 Take a look at : https://github.com/tititopher68-dev2/Marlin/tree/new_feature

Wow man @tititopher68-dev, can't believe you just coded this feature after like 8 hours from @jumper-11 request. There you go! Nice job! xD

**Reply by Cal-LF** [reply]
@tititopher68-dev Trying the latest commit and have compile errors immediately. Configs are for the E3V2 manual mesh 3x3 v427.

**Comment by Cal-LF** [comment]
@tititopher68-dev I tried again this AM. This is what I am getting even with no change in configs

![error](https://user-images.githubusercontent.com/76499230/166688468-bb2505db-b9d7-4a0c-8814-30e9543d580e.jpg)
.

**Reply by SikVac** [reply]
> @Cal-LF it's not a firmware issue but a VSCode issue, VSCode has an update for platformio since 2 days....

I have the same error for E3V2, v422, RET6, UBL-BLTouch-15x15. VSCode is updated !!!

**Reply by jumper-11** [reply]
Thanks for the change.
I have already compiled the BL_Touch5x5 version for the 4.2.2 board. E3v2, but the BL menu did not appear - has anyone tested it? The compilation was successful.
For sure?
Which file should I compile? Configuration.h or Configurations_adv.h?

**Reply by jumper-11** [reply]
I just checked at home.
There is a sort option, but it shows from "oldest" and not from the newest.
I am asking for improvement.

**Reply by jumper-11** [reply]
I use good sources.
Sorting doesn't work on 2 different devices.
E3 v3 board 4.2.2 and motherboard 4.2.7

**Reply by jumper-11** [reply]
I can see there are new options in the menu and on the screen, but sorting is not working.
I have uploaded a new file for testing on the card and it is at the very bottom.

**Reply by jumper-11** [reply]
How about sorting files, has anyone else verified it? For me, the sorting from the newest at the top does not work on 2 printers.
